accounts/ethstore/src/account/safe_account.rs

```diff
@@ -85,35 +85,32 @@ impl SafeAccount {
 		let crypto = Crypto::from(json.crypto);
 		let address = match (password, &json.address) {
 			(None, Some(json_address)) => json_address.into(),
-			(None, None) => Err(Error::Custom(
-				"This keystore does not contain address. You need to provide password to import it".into()))?,
+			(None, None) => return Err(Error::Custom(
+				"This keystore does not contain address. You need to provide password to import it".into())),
 			(Some(password), json_address) => {
 				let derived_address = KeyPair::from_secret(
-					crypto.secret(&password).map_err(|_| Error::InvalidPassword)?
+					crypto.secret(password).map_err(|_| Error::InvalidPassword)?
 				)?.address();
 
-				match json_address {
-					Some(json_address) => {
-						let json_address = json_address.into();
-						if derived_address != json_address {
-							warn!("Detected address mismatch when opening an account. Derived: {:?}, in json got: {:?}",
-								derived_address, json_address);
-						}
-					},
-					_ => {},
+				if let Some(json_address) = json_address {
+					let json_address = json_address.into();
+					if derived_address != json_address {
+						warn!("Detected address mismatch when opening an account. Derived: {:?}, in json got: {:?}",
+							derived_address, json_address);
+					}
 				}
 				derived_address
 			}
 		};
 
-		Ok(SafeAccount {
+		Ok(Self {
 			id: json.id.into(),
 			version: json.version.into(),
 			address,
 			crypto,
 			filename,
-			name: json.name.unwrap_or(String::new()),
-			meta: json.meta.unwrap_or("{}".to_owned()),
+			name: json.name.unwrap_or_else(String::new),
```

Comment:
I would prefer `unwrap_or_default` but not a big deal

Reply:
done